src/main/webapp/app/overview/discussion-section/discussion-section.component.ts

```diff
@@ -185,13 +185,19 @@ export class DiscussionSectionComponent extends CourseDiscussionDirective implem
                         return;
                     }
 
-                    const contextFilter = this.channel?.id
-                        ? { conversationId: this.channel.id }
-                        : {
-                              exerciseIds: this.exercise?.id !== undefined ? [this.exercise.id] : undefined,
-                              lectureIds: this.lecture?.id !== undefined ? [this.lecture.id] : undefined,
-                          };
-                    this.metisService.getFilteredPosts(contextFilter);
+                    if (this.channel?.id) {
+                        const contextFilter = { conversationId: this.channel.id };
+                        const channelDTO = new ChannelDTO();
+                        channelDTO.isCourseWide = true;
+                        this.metisService.getFilteredPosts(contextFilter, true, channelDTO);
+                    } else {
+                        const contextFilter = {
+                            exerciseIds: this.exercise?.id !== undefined ? [this.exercise.id] : undefined,
```

Comment:
```suggestion
                            exerciseIds: this.exercise?.id ? [this.exercise.id] : undefined,
```

Reply:
In theory, the exercise id could be 0, but in practice this will not happen. So I guess it is fine to change it